client.go

```diff
@@ -68,11 +68,10 @@ func (c Client) Send(filename string, mode string) (io.ReaderFrom, error) {
 		s.opts["blksize"] = strconv.Itoa(c.blksize)
 	}
 	n := packRQ(s.send, opWRQ, filename, mode, s.opts)
-	addr, err := s.sendWithRetry(n)
+	err = s.sendWithRetry(n)
 	if err != nil {
 		return nil, err
 	}
-	s.addr = addr
```

Comment:
Yeah I hadn't realized that sender was reused between both client and server, thus why the tests failed.  It doesn't look like Go exposes a mechanism to connect an existing unconnected UDP socket so DialUDP won't work.  I've actually refactored all this code to perform the validation and pass the test cases (yay!).

I've just learned Go over the last couple weeks and didn't know how to invoke the built in test framework.  I've figured that out now.  Will submit a new PR that passes the tests soon.